### ISSUE TITLE
Starting up with an .Rprofile

### ISSUE THREAD

**Issue by grimmjulian** [jalvesaq/Nvim-R#721]
Hello,

I am using an .Rprofile in my Home directory to set some options like browser and default CRAN repo as well as setting up the languageserver.
This is my .Rprofile:
```
formals(quit)$save <- "no"
formals(q)$save <- "no"
options(
  languageserver.server_capabilities =
    list(completionProvider = FALSE, completionItemResolve = FALSE)
)
options(browser = "firefox")
options(repos = c(CRAN = "https://cloud.r-project.org/"))
```
This does not change anything coding related and worked fine, till I started using renv now.

The problem arises because renv writes a project specific .Rprofile and then my Home-.Rprofile will not get sourced. I could add a line in my Project-.Rprofile that it should source my Home-.Rprofile, but I think it would be bad style, because the Home-.Rprofile would not be in the project repository, but sourced.

Is there a way to set this options inside of Nvim-R at startup, so that I do not need an .Rprofile?

**Comment by jalvesaq** [comment]
> Is there a way to set this options inside of Nvim-R at startup, so that I do not need an .Rprofile?

Not yet.

You could try this in your `~/.Rprofile`:

```r
if (Sys.getenv("NVIMR_TMPDIR") == "") {
    formals(quit)$save <- "no"
    formals(q)$save <- "no"
    options(
            languageserver.server_capabilities =
                list(completionProvider = FALSE, completionItemResolve = FALSE)
    )
    options(browser = "firefox")
    options(repos = c(CRAN = "https://cloud.r-project.org/"))
}
```

But if editing the `~/.Rprofile` is not an option in your case, I could change the option `R_after_start` to accept lines that don't begin with either `!` or `:` and add these lines to `start_options_utf8.R`, which is saved by Nvim-R in its temporary directory. So, in addition to running shell and Vim commands after R startup, users would be able to make R run some code too.

**Comment by grimmjulian** [comment]
Thanks for the fast answer! I was just wondering, because in RStudio these settings are already set.
For now your proposal for the .Rprofile will work as I only use Nvim as IDE in private. I will postpone then switching to Nvim in my job a little bit :)

For your proposal for the option, I am to inexperienced right now and do not really understand what you are saying. I will look into it during the week and come back to you.

Thanks again for the fast answer!

**Comment by jalvesaq** [comment]
I did a test here, and we can't change the "browser" option in RStudio during `nvimcom` loading. Certainly, RStudio changes the "browser" option _after_ reading the `~/.Rprofile` and loading all default packages. So, the use of `R_after_start` option would not work for this.

Regarding `renv`, I don't use it but its documentation says:

> If you want to set and persist these options across multiple projects, it is recommended that you set them in a a startup .Renviron file; e.g. in your own ~/.Renviron, or in the R installation's etc/Rprofile.site file. (https://rstudio.github.io/renv/reference/config.html)

**Comment by grimmjulian** [comment]
I do not use .Renviron files but I do not think you can set general options there. 

I realised another problem, why it would be good to have some R script before hand.
For the language server to work, I need to set these options. Even if I can set these options via .Renviron, this leaves another renv problem: the "languageserver" package is not installed in a new renv library and therefore the lsp client cannot start. Before starting the "R programming", I have to check, if the languageserver package is installed and install if not.

I changed the .Rprofile in my Home directory to do this now, but because of R only reading the top most .Rprofile, this will be probably ignored sometimes when there are project specific .Rprofiles. If there is no other way, I will rewrite these, but I really hoped for some way to automate this.

**Comment by jalvesaq** [comment]
I can't help because I don't use `renv`. But if you know of specific changes in Nvim-R that are needed to solve your problem, I may try to implement them.

**Comment by grimmjulian** [comment]
Ok, so basically there are two questions that I have that I would remove from the .Rprofile:

1. How can I make sure, the language server package is installed? As this is probably more of an LSP problem, I will ask there, maybe they have a solution?
1. How would it be possible to set the options (e.g. browser, repos...) for R similar to RStudio?

If I understand you correctly, right now this two things cannot be solved in Nvim-R but only through .Rprofile and .Renviron. If this is correct, the issue can be closed till I maybe have a better grasp of the problem and Nvim-R. For now, using a project specific .Rprofile that includes the nvim part as you suggested works and I can use it for my private programminging, even if I do not like the idea of having "IDE settings" in my .Rprofile.

**Comment by jalvesaq** [comment]
`R_after_start` now includes strings prefixed with `R:` as commands to be executed during nvimcom loading. This will not fix the "browser" option if you are running RStudio, though. See https://community.rstudio.com/t/changing-the-default-web-browser-used-in-rstudio/23653

**Comment by grimmjulian** [comment]
Sorry for the late answer. I tried it out and it seems to work. Because it was a real hassle to get the script with quotationsmarks to work, I simply wrote everything in an R file and sourced it.
Thanks!
